src/Features/CSharp/Portable/UseSimpleUsingStatement/UseSimpleUsingStatementDiagnosticAnalyzer.cs

```diff
@@ -96,6 +99,18 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
             }
 
             var cancellationToken = context.CancellationToken;
+
+            // Converting a using-statement to a using-variable-declaration will cause the using's
+            // variables to now be pushed up to the parent block's scope. This is also true for any
+            // local variables in the innermost using's block. These may then collide with other
+            // variables in the block, causing an error.  Check for that and bail if this happens.
+            if (CausesVariableCollision(
```

Comment:
Would it be possible to use any of the [ISemanticFactsService.GenerateUniqueName](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/LanguageServices/SemanticsFactsService/ISemanticFactsService.cs,107) helpers? Possibly using a speculative semantic model where the using statement being analyzed is itself removed and the using statement variable name is the baseName passed to the API?

Reply:
This does use one of the core helpers.  Specifically, the GetExistingSymbols helper (which GenerateUniqueName depends upon.

Reply:
> Possibly using a speculative semantic model where the using statement being analyzed is itself removed and the using statement variable name is the baseName passed to the API?

Note: you can't use speculative models for things like actual removal.  You'd need a real model.  I'm a little wary of going that route because of hte perf cost here.  i.e. you end up having to take the perf hit of binding the changed semantic model, rather than just doing the check yourself.  the former is more accurate (you can just use the compiler), but the latter is much faster.  

We tend to avoid the former due to the high costs (though some features occasionally do it if the check is so complex to be too difficult to write ourselves).

Reply:
Thanks, current approach seems fine as well. I was just wondering if we can avoid symbol matching and collision checking here.